modules/machine-lifecycle-hook-deletion.adoc

```diff
@@ -0,0 +1,221 @@
++
+Until all `preTerminate` lifecycle hooks are satisfied, the machine status condition `Terminable` is set to `False`.
+. There are no unresolved `preTerminate` lifecycle hooks and the machine status condition `Terminable` is set to `True`.
+. The machine controller removes the instance from the infrastructure provider.
+. The machine controller deletes the `Node` object.
+
+[id="machine-lifecycle-hook-deletion-uses_{context}"]
+== Example Operator use cases
+
+Operators can use lifecycle hooks for the machine deletion phase to modify the machine deletion process.
+//I'm not totally sure if it makes sense to include this in prod docs. It seems like something third-party Operator devs would want to know, rather than end-users (aside from the quorum protection stuff, but I have included that separately). If the content is solid, I can talk to the Operators writers about whether they have ideas on where it might live.
+
+Implementing custom draining logic:: An Operator can use a `preDrain` lifecycle hook to replace the machine controller draining logic with a different draining controller. By replacing the draining logic, the Operator would have more flexibility and control over the lifecycle of the workloads on each node.
++
+For example, the machine controller drain libraries do not support ordering, but a custom drain provider could provide this functionality. By using a custom drain provider, an Operator could prioritize moving mission-critical applications before draining the node to ensure that service interruptions are minimized in cases where cluster capacity is limited.
+
+Verifying storage detachment:: An Operator can use a `preTerminate` lifecycle hook to ensure that storage that is attached to a machine is detached before the machine is removed from the infrastructure provider.
+
+Improving log reliability:: After a node is drained, the log exporter daemon requires some time to synchronize logs to the centralized logging system.
++
+A logging Operator can use a `preTerminate` lifecycle hook to add a delay between when the node drains and when the machine is removed from the infrastructure provider. This delay would provide time for the Operator to ensure that the main workloads are removed and no longer adding to the log backlog. When no new data is being added to the log backlog, the log exporter can catch up on the synchronization process, thus ensuring that all application logs are captured.
+
+Proactively replacing machines:: An Operator can use a `preTerminate` lifecycle hook to ensure that a replacement machine is successfully created and joined to the cluster before removing the instance of a deleted machine. This can mitigate the impact of disruptions during machine replacement or of replacement instances that do not initialize promptly.
+//The enhancement example is a control plane machine, but seems could apply to any type of machine if needed.
```

Comment:
You'd probably want a preDrain on this case else you're moving workloads onto ??? before you've waited for that new machine to com up

Reply:
I wondered about that - it's `preTerminate` in the enhancement (Story 1) but it sounded like what quorum protection is doing and that's `preDrain`. Thanks! :)